### PR TITLE
Feature/sign in

### PR DESCRIPTION
**Pull request by Kaskyi**

### CODE REVIEW

**Review by Sumragen — APPROVED** [review]

File: packages/components/parallax/mouse-move-parallax.directive.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+@Directive({
+    selector: '[xmMouseMoveParallax]',
+})
+export class MouseMoveParallaxDirective {
+
+    constructor(private el: ElementRef<HTMLImageElement>) {
+    }
+
+    @HostListener('document:mousemove', ['$event'])
+    public onWindowResize(e: MouseEvent): void {
```
Comment (avchugaev) [inline packages/components/parallax/mouse-move-parallax.directive.ts:12]: Please check the method name. Mayby it should be `onDocumentMouseMove()`?

File: packages/components/parallax/mouse-move-parallax.directive.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    }
+
+}
+
+
+@NgModule({
+    exports: [MouseMoveParallaxDirective],
+    declarations: [MouseMoveParallaxDirective],
+})
+export class MouseMoveParallaxModule {
```
Comment (avchugaev) [inline packages/components/parallax/mouse-move-parallax.directive.ts:29]: Please eject the module class into a separate file.

File: packages/components/particles/canvas-with-particles/canvas-with-particles.component.ts [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    }
+
+}
+
+
+@NgModule({
+    exports: [CanvasWithParticlesComponent],
+    declarations: [CanvasWithParticlesComponent],
+})
+export class CanvasWithParticlesModule {
```
Comment (avchugaev) [inline packages/components/particles/canvas-with-particles/canvas-with-particles.component.ts:73]: Eject to separate file.

File: packages/components/particles/line-trail-particle.ts [hunk 4]
```diff
@@ -0,0 +1,168 @@
+import { clone } from 'lodash';
+
+export const randomInt = (min, max) => Math.floor(Math.random() * (max - min + 1)) + min;
```
Comment (avchugaev) [inline packages/components/particles/line-trail-particle.ts:3]: Don't export if it won't be used elsewhere else.

File: packages/components/particles/line-trail-particle.ts [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+    public x: number = this.options.x;
+    public y: number = this.options.y;
+    public color: Color = clone(this.options.color);
+    public speed: number = this.options.speed;
+
+    constructor(public options: LineTrailParticleOptions) {
+    }
+}
+
+export class LiveTimeColorScript {
```
Comment (avchugaev) [inline packages/components/particles/line-trail-particle.ts:38]: Please eject classes into corresponding classes or remove the `export` keyword.

File: packages/components/sign-in/sign-in-form/xm-sign-in-form.component.ts [hunk 6]
```diff
@@ -0,0 +1,51 @@
+import { Component, EventEmitter, Input, Output } from '@angular/core';
+import { FormControl, FormGroup } from '@angular/forms';
+import { SignInForm, SignInService } from './sign-in.service';
+import { XmEmailControlOptions, XmPasswordControlOptions } from '@xm-ngx/components/text';
+import { Translate } from '@xm-ngx/translation';
+import { Observable } from 'rxjs/Observable';
+import { Subject } from 'rxjs/Subject';
+
+export interface XmSignInFormOptions {
```
Comment (avchugaev) [inline packages/components/sign-in/sign-in-form/xm-sign-in-form.component.ts:9]: Could it be ejected into a separate file?

File: packages/components/text/xm-password-control.ts [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+import { ReactiveFormsModule } from '@angular/forms';
+import { MatFormFieldModule } from '@angular/material/form-field';
+import { MatIconModule } from '@angular/material/icon';
+import { MatInputModule } from '@angular/material/input';
+import { ControlErrorModule } from '@xm-ngx/components/control-error';
+import { NgFormAccessor } from '@xm-ngx/components/ng-accessor';
+import { IControlFn } from '@xm-ngx/dynamic';
+import { Translate, XmTranslationModule } from '@xm-ngx/translation';
+
+export interface XmPasswordControlOptions {
```
Comment (avchugaev) [inline packages/components/text/xm-password-control.ts:11]: Eject classes and interfaces into their own files, please

**Review by avchugaev — APPROVED** [review]

File: packages/components/particles/line-trail-particle.ts (hunk 4 above)
Comment (Kaskyi) [reply packages/components/particles/line-trail-particle.ts:3]: it used by randomValue

File: packages/components/parallax/mouse-move-parallax.directive.ts (hunk 2 above)
Comment (Kaskyi) [reply packages/components/parallax/mouse-move-parallax.directive.ts:29]: +

File: packages/components/parallax/mouse-move-parallax.directive.ts (hunk 1 above)
Comment (Kaskyi) [reply packages/components/parallax/mouse-move-parallax.directive.ts:12]: \+

File: packages/components/particles/canvas-with-particles/canvas-with-particles.component.ts (hunk 3 above)
Comment (Kaskyi) [reply packages/components/particles/canvas-with-particles/canvas-with-particles.component.ts:73]: \+

File: packages/components/text/xm-password-control.ts (hunk 7 above)
Comment (Kaskyi) [reply packages/components/text/xm-password-control.ts:11]: \+

File: packages/components/sign-in/sign-in-form/xm-sign-in-form.component.ts (hunk 6 above)
Comment (Kaskyi) [reply packages/components/sign-in/sign-in-form/xm-sign-in-form.component.ts:9]: \+

**Pull request merged**